src/service/xhr-impl.js

```diff
@@ -104,21 +104,29 @@ export class Xhr {
   }
 
   /**
-   * Performs the final initialization and requests the fetch. It does three
-   * main things: (1) It adds "__amp_source_origin" URL parameter with source
-   * origin; (2) It verifies "AMP-Access-Control-Allow-Source-Origin" if it's
-   * returned in the response; and (3) It requires
-   * "AMP-Access-Control-Allow-Source-Origin" to be present in the response
-   * if the `init.requireAmpResponseSourceOrigin = true`.
-   * USE WITH CAUTION: setting ampCors false disables AMP source origin check
+   * Performs the final initialization and requests the fetch. It does two
+   * main things:
+   * - It adds "__amp_source_origin" URL parameter with source origin
+   * - It verifies "AMP-Access-Control-Allow-Source-Origin" in the response
+   * USE WITH CAUTION: setting ampCors to false disables AMP source origin check
    * but allows for caching resources cross pages.
+   * @deprecated requireAmpResponseSourceOrigin is deprecated. It defaults to
+   *   true. Use "ampCors: false" to disable AMP source origin check.
    *
    * @param {string} input
    * @param {!FetchInitDef=} init
    * @return {!Promise<!FetchResponse>}
    * @private
    */
   fetchAmpCors_(input, init = {}) {
+    if (init.requireAmpResponseSourceOrigin === true) {
+      dev().error('XHR',
+          'requireAmpResponseSourceOrigin is deprecated, use ampCors instead');
+    }
+
+    if (init.requireAmpResponseSourceOrigin === undefined) {
```

Comment:
Yes, you must be conscious about it. But renaming APIs in core definitely causes such issues.

We try to avoid skew in as many places as we can, but it can absolutely happen.